### ISSUE TITLE
Cross layer filtering on WFS takes too much on the demo

### ISSUE THREAD

**Issue by offtherailz** [geosolutions-it/lhtac-webgis#66]
The select points takes too much to response. 
![image](https://cloud.githubusercontent.com/assets/1279510/20258070/022aece2-aa4e-11e6-847d-b0f5b385f616.png)

The data is in shape files. There should be some optimizations to perform cross layer filtering directly on db

**Comment by offtherailz** [comment]
There are 2 different problems : 
1) The proxy, The request with the  takes too much time to get the response.
![image](https://cloud.githubusercontent.com/assets/1279510/20258815/bcd57410-aa51-11e6-8c91-8f58c7bcc835.png) 
![image](https://cloud.githubusercontent.com/assets/1279510/20258859/f5d9ee26-aa51-11e6-9841-b98495d2fb4f.png)

2) selecting a big area takes to much to response even if performed directly to the server: 

example:
```
<wfs:GetFeature service="WFS" version="1.1.0" xmlns:gml="http://www.opengis.net/gml" xmlns:wfs="http://www.opengis.net/wfs" xmlns:ogc="http://www.opengis.net/ogc" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.1.0/wfs.xsd"><wfs:Query typeName="lhtac:web2014all_mv" srsName="EPSG:4326"><ogc:Filter><ogc:And><ogc:Intersects><ogc:PropertyName>geom3857</ogc:PropertyName><gml:Polygon srsName="EPSG:4326"><gml:exterior><gml:LinearRing><gml:posList>-117.22412109375 47.724544549099676 -117.22412109375 48.21735290928554 -116.20239257812499 48.21735290928554 -116.20239257812499 47.724544549099676 -117.22412109375 47.724544549099676</gml:posList></gml:LinearRing></gml:exterior></gml:Polygon></ogc:Intersects><ogc:Intersects><ogc:PropertyName>geom3857</ogc:PropertyName><ogc:Function name="collectGeometries"><ogc:Function name="queryCollection"><ogc:Literal>lhtac:itd_districts</ogc:Literal><ogc:Literal>the_geom</ogc:Literal><ogc:Literal>ITD_Dist_n IN ('1')</ogc:Literal></ogc:Function></ogc:Function></ogc:Intersects></ogc:And></ogc:Filter></wfs:Query></wfs:GetFeature>
```

![image](https://cloud.githubusercontent.com/assets/1279510/20258942/64d27d8e-aa52-11e6-9866-7e8a7a47cfc5.png)

**Comment by aaime** [comment]
Note: cross layer filtering has presently no DBMS based optimization, although one could be added if we wanted to. It just prevents a client round trip, it's a poor man solution: http://www.geo-solutions.it/blog/developer-corner-cross-layer-filtering-coming-down-to-geoserver/

**Comment by offtherailz** [comment]
Than you for your feedback @aaime .
During the tests I improved the client to avoid the usage of the proxy, I noticed it
this was the real bottleneck. 
Now the requests take less that 3-4 seconds to be performed (almost all content download).
![image](https://cloud.githubusercontent.com/assets/1279510/20262715/9eb1682e-aa63-11e6-8160-4d148f21df09.png)

So the cross origin filtering is not a bottleneck for this specific application. 

The only thing left to do is to limit the number of features downloaded. If they are too much, it can decrease the browser performances.
